indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -177,13 +178,23 @@ private boolean createAndStartSupervisorInternal(SupervisorSpec spec, boolean pe
       return false;
     }
 
-    Supervisor supervisor = spec.createSupervisor();
-    supervisor.start(); // try starting the supervisor first so we don't persist a bad spec
-
     if (persistSpec) {
       metadataSupervisorManager.insert(id, spec);
     }
 
+    Supervisor supervisor = null;
+    try {
+      supervisor = spec.createSupervisor();
+      supervisor.start();
+    }
+    catch (Exception e) {
+      // Supervisor creation or start failed write tombstone only when trying to start a new supervisor
```

Comment:
@himanshug to be safe, we should probably make a call to consumer.close() in KafkaSupervisor.start() if the Kafka consumer was instantiated but an exception was thrown afterwards; @pjain1 do you mind adding this?